src/ripple/ledger/ReadView.h

```diff
@@ -285,8 +310,28 @@ class DigestAwareReadView
     digest (key_type const& key) const = 0;
 };
 
-// For now
-using BasicView = ReadView;
+/** Run a functor on each SLE in a ReadView starting from the key start,
+    as long as the functor returns true.
+ */
+template <class Functor>
+void forEachSLE(ReadView const& view, Functor func, uint256 const& start = {})
```

Comment:
It only visits every SLE start is empty AND if the functor returns true each time.

We use this in two separate handlers but we'll end up using this every time we want to iterate a subrange of the SLEs.  Given that iterating is a non-trivial block of code...